### ISSUE TITLE
Crackling SoundBlaster sound - bisected

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#2386]
**Describe the bug**
I have been getting terrible crackling/static SB audio from DOSBox-X for quite some time. Switching to vanilla DOSBox or dosbox-staging and all is fine.

This has been going on for a while, but I have been too busy of late. But now I decided to bisect the issue. I had to go all the way back to 0.83.2 to get decent SB audio.

The strange thing is, that my bisect came back to this commit:

```
1af63ad365e680efa5ce1c1baf327d7e129d448f is the first bad commit
commit 1af63ad365e680efa5ce1c1baf327d7e129d448f
Author: rderooy <robert.de.rooy@gmail.com>
Date:   Sat Jun 6 11:44:07 2020 +0200

    FluidSynth - Change Linux default driver from Jack to PulseAudio
    PulseAudio is more likely to work by default, if you want Jack
    you can still set it in the config.

 src/gui/midi_synth.h | 4 ++++
 1 file changed, 4 insertions(+)
```

Which should have nothing to do with SoundBlaster. But a quick test (with prince of persia) confirms it. Using the latest nightly build:
```
dosbox-x -set mididevice=none PRINCE.EXE
```

By setting mididevice to none, the SB sound is fixed.

Any idea what may be going on?

**Comment by Wengier** [comment]
Is your "mididevice" option sets to "fluidsynth"? What if you set e.g. "mididevice=default"? Meanwhile, the change in the said commit seems to only affect the Linux platform.

**Comment by grapeli** [comment]
This has 100% nothing to do with the code or the dosbox-x configuration.

If you hear "effects" like in the sample you posted in this [thread](https://github.com/joncampbell123/dosbox-x/issues/1810#issuecomment-678984993) (pop2_gm.zip), then your system pulseaudio sound is misconfigured. Look for the reason there.
https://wiki.archlinux.org/index.php/PulseAudio

By the way, wouldn't it be better to use the SDL2 driver for fluidsynth in your case?
Provided that your fluidsynth library makes it available.
https://github.com/FluidSynth/fluidsynth/blob/master/src/drivers/fluid_sdl2.c

**Comment by rderooy** [comment]
Yes this is on Linux.

I just quickly confirmed that 0.83.2 has the same issue when I enable pulseaudio for fluidsysnth.
```
dosbox-x -set fluid.driver=pulseaudio PRINCE.EXE
```

**Comment by rderooy** [comment]
To give a bit more background... About a month ago, I replaced an ageing Intel i5-4000 series based system with a brand new AMD Ryzen 5 5600X system.

It seems this issue effects the new AMD system, but does not effect the old Intel system. I just double checked, and the old Intel system, which I still have sitting here, indeed does not exhibit the issue.

So this is either a timing issue on the faster machine, or a bug in the AMD sound driver?

I checked this with connecting a USB speaker to the AMD system, and Prince of Persia played perfectly. So that lead me to think there is some issue with the AMD sound driver.

Googling, I found this old 2014 issue https://askubuntu.com/questions/405071/static-and-crackling-in-my-hdmi-audio

Thinking it would never work, since they must have fixed whatever issue is causing it, I tried it nonetheless. And guess what, the audio corruption is gone.
